lnbits/middleware.py

```diff
@@ -210,3 +210,18 @@ async def block_allow_ip_middleware(request: Request, call_next):
         return await call_next(request)
 
     app.middleware("http")(block_allow_ip_middleware)
+
+
+def add_first_install_middleware(app: FastAPI):
+    @app.middleware("http")
+    async def first_install_middleware(request: Request, call_next):
+        if request.url.path != "/api/v1/auth/first_install":
+            if (
+                settings.first_install
+                and request.url.path != "/first_install"
+                and not request.url.path.startswith("/static")
+            ):
+                return RedirectResponse("/first_install")
+        return await call_next(request)
+
+    app.middleware("http")(first_install_middleware)
```

Comment:
We should be careful with middlewares. Their logic is executed for each HTTP request, so they can increase the response time.
In this particular case, the logic is relevant only once in the life of the installation.

Reply:
the if could be combined but i dont think a check in memory `settings.first_install ` is expensive